project_report/tex/analysis.tex

```diff
@@ -15,3 +15,9 @@ \subsubsection{Sequence Diagrams}
 \subsubsection{Operation Contracts}
 
 \subsubsection{Updated UML Class Diagram}
+\begin{figure}[ht]
+\centering 
+\includegraphics[scale=0.3]{images/diagrams/updated_UML_Class_Diagram.drawio.png}
```

Comment:
I think there is some spelling mistakes? and I think the correct way to spell them is after -> 
SubscribtionHandler -> SubscriptionHandler
machineDataColllector -> machineDataCollector

![image](https://user-images.githubusercontent.com/55736468/96860359-f1bd5600-1462-11eb-9c46-5435b7483980.png)
Can the return type be one of these? that is pointed to? 
or that does not matter?

Reply:
machine state is a separate thing it is used to translate the current machine state